askama_parser/src/lib.rs

```diff
@@ -487,3 +511,77 @@ fn filter(i: &str, level: Level) -> ParseResult<'_, (&str, Option<Vec<Expr<'_>>>
     ))(i)?;
     Ok((i, (fname, args)))
 }
+
+/// The goal of this function is to reduce the path length based on the `base` argument
+/// (generally the path where the program is running into). For example:
+///
+/// ```text
+/// current dir: /a/b/c
+/// path:        /a/b/c/d/e.txt
+/// ```
+///
+/// `strip_common` will return `d/e.txt`.
+fn strip_common(base: &Path, path: &Path) -> String {
+    let path = match path.canonicalize() {
+        Ok(path) => path,
+        Err(_) => return path.display().to_string(),
+    };
+    let mut components_iter = path.components().peekable();
+
+    for current_path_component in base.components() {
+        match components_iter.peek() {
+            Some(path_component) => {
+                if current_path_component != *path_component {
+                    break;
+                }
+                components_iter.next();
+            }
+            None => return path.display().to_string(),
+        }
```

Comment:
Nit: looks like a case for `let-else`?

Reply:
We don't assign here so `if let` maybe?

Reply:
I meant this:

```rust
let Some(path_component) = components_iter.peek() else {
    return path.display().to_string();
}
```

Reply:
... It's obvious when you see it. :sweat_smile: